### PR TITLE
Bugfix MTE-3146 Firefox iOS tests on Github Actions

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-3146)

## :bulb: Description
I have tried to mitigate the limitations of the Github Actions runners by the following:
* Standardize timeout (`TIMEOUT`) to 20 seconds. `TIMEOUT_LONG` is 60 seconds. Use default timeout unless longer timeout is needed.
* Ping the websites require by the test just to see if they are up.
* Use `mozWait` instead of `XCTAssert` to confirm the existence.

In addition, here are some files for debugging:
* xcodebuild.log
* Junit XML files

Sample run: https://github.com/mozilla-mobile/firefox-ios/actions/runs/10065020448

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: .github/workflows/firefox-ios-ui-tests.yml [hunk 1]
```diff
@@ -40,7 +40,8 @@ jobs:
                   -scheme ${{ env.xcodebuild_scheme }} \
                   -target ${{ env.xcodebuild_target }} \
                   -derivedDataPath ~/DerivedData \
-                  -destination 'platform=iOS Simulator,name=${{ env.ios_simulator_default }},OS=${{ env.ios_version }}'
+                  -destination 'platform=iOS Simulator,name=${{ env.ios_simulator_default }},OS=${{ env.ios_version }}' \
+                  COMPILER_INDEX_STORE_ENABLE=NO CODE_SIGN_IDENTITY= CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO ARCHS="arm64"
```
Comment (clarmso) [inline .github/workflows/firefox-ios-ui-tests.yml:44]: Without `ARCHS="arm64"`, some seemingly `x64_64` tools has been used in the compilation process. I'm not sure if such tools affected the binary. I'd like to use M1 binaries on the safe side.

File: .github/workflows/firefox-ios-ui-tests.yml [hunk 2]
```diff
@@ -70,7 +71,29 @@ jobs:
...
+          curl https://theguardian.com > /dev/null
+          curl https://www.reddit.com > /dev/null
+          curl http://www.iana.org/domains/example > /dev/null
+          curl https://www.twitter.com > /dev/null
+          curl https://www.youtube.com > /dev/null
+          curl https://www.instagram.com > /dev/null
+          curl https://www.euronews.com > /dev/null
+          curl https://www.fakespot.com > /dev/null
+          curl https://www.badssl.com > /dev/null
+          curl https://accounts.google.com > /dev/null
```
Comment (clarmso) [inline .github/workflows/firefox-ios-ui-tests.yml:96]: If the URLs are available, I've no idea why the tests have trouble loading some of the websites intermittently.

File: .github/workflows/firefox-ios-ui-tests.yml [hunk 3]
```diff
@@ -203,81 +238,106 @@ jobs:
...
-              run: |
-                exit ${{ steps.run-tests.outcome == 'success' && '0' || '1' }}
\ No newline at end of file
+      name: Run full functional tests
+      runs-on: macos-14
+      needs: compile
+      strategy:
+        fail-fast: false
+        matrix:
+          ios_simulator: ['iPhone 15 Pro Max']
```
Comment (clarmso) [inline .github/workflows/firefox-ios-ui-tests.yml:247]: It takes too long to run the iPad tests in the same workflow. Github Actions killed the unfinished job.

File: firefox-ios/firefox-ios-tests/Tests/Smoketest1.xctestplan [hunk 4]
```diff
@@ -174,6 +174,7 @@
         "PrivateBrowsingTestIpad",
         "PrivateBrowsingTestIpad\/testClosePrivateTabsOptionClosesPrivateTabsShortCutiPad()",
         "PrivateBrowsingTestIpad\/testiPadDirectAccessPrivateModeBrowserTab()",
+        "ReadingListTests",
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/Smoketest1.xctestplan:177]: I will re-enable ReadingListTests.

**Comment by isabelrios** [comment]
@clarmso not sure this PR is right, adding those other commits...

**Comment by clarmso** [comment]
> @clarmso not sure this PR is right, adding those other commits...

😞 Must be from a merge commit again. Let me cherry pick the changes.

**Pull request closed without merging**